src/main/java/com/jcabi/aspects/aj/UpdateMethodCacher.java

```diff
@@ -0,0 +1,178 @@
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.aspects.aj;
+
+import com.jcabi.aspects.Loggable;
+import com.jcabi.log.VerboseRunnable;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * For updating cache and cleaning expired cache.
+ * @author Peng Chuanjiang (pengchuanjiang@souyidai.com)
+ * @version $Id$
+ */
+@SuppressWarnings("PMD.DoNotUseThreads")
+public final class UpdateMethodCacher {
+
+    /**
+     * Calling tunnels.
+     * @checkstyle LineLength (2 lines)
+     */
+    private final transient ConcurrentMap<MethodCacher.Key, MethodCacher.Tunnel> tunnels;
+
+    /**
+     * Service that cleans cache.
+     */
+    private final transient ScheduledExecutorService cleaner =
+        Executors.newSingleThreadScheduledExecutor(
+            new NamedThreads(
+                "cacheable-clean",
+                "automated cleaning of expired @Cacheable values"
+            )
+        );
+
+    /**
+     * Save the keys which need update.
+     */
+    private final transient BlockingQueue<MethodCacher.Key> updatekeys;
+
+    /**
+     * Service that update cache.
+     */
+    private final transient ScheduledExecutorService updater =
+        Executors.newSingleThreadScheduledExecutor(
+            new NamedThreads(
+                "cacheable-update",
+                "async update of expired @Cacheable values"
+            )
+        );
+
+    /**
+     * Public ctor.
+     * @param tls ConcurrentMap
+     * @param ukeys BlockingQueue
+     */
+    public UpdateMethodCacher(
+        final ConcurrentMap<MethodCacher.Key, MethodCacher.Tunnel> tls,
+        final BlockingQueue<MethodCacher.Key> ukeys) {
+        this.tunnels = tls;
+        this.updatekeys = ukeys;
+    }
+
+    /**
+     * Staring two threads(cheaner and updater) are used for
+     * updating cache and cleaning expired cache.
+     */
+    public void start() {
+        this.cleaner.scheduleWithFixedDelay(
+            new VerboseRunnable(
+                new Runnable() {
+                    @Override
+                    public void run() {
+                        UpdateMethodCacher.this.clean();
+                    }
+                }
+            ),
+            1L, 1L, TimeUnit.SECONDS
+        );
+        this.updater.schedule(
+            new VerboseRunnable(
+                new Runnable() {
+                    @Override
+                    public void run() {
+                        UpdateMethodCacher.this.update();
+                    }
+                }
+            ),
+            0L, TimeUnit.SECONDS
```

Comment:
@54uso so it is a one-shot action? it doesn't execute again after a time interval?